client/src/pages/EditProfile/EditProfile.tsx

```diff
@@ -0,0 +1,99 @@
+import CssBaseline from '@material-ui/core/CssBaseline';
+import Paper from '@material-ui/core/Paper';
+import Box from '@material-ui/core/Box';
+import Grid from '@material-ui/core/Grid';
+import { FormikHelpers } from 'formik';
+import Typography from '@material-ui/core/Typography';
+import useStyles from './useStyles';
+import updateProfile from '../../helpers/APICalls/updateProfile';
+import changePassword from '../../helpers/APICalls/changePassword';
+import EditProfileForm from './EditProfileForm/EditProfileForm';
+import ChangePasswordForm from './ChangePasswordForm/ChangePasswordForm';
+import { useAuth } from '../../context/useAuthContext';
+import { useSnackBar } from '../../context/useSnackbarContext';
+import { CircularProgress } from '@material-ui/core';
+import { useHistory } from 'react-router-dom';
+
+export default function EditProfile(): JSX.Element {
+  const classes = useStyles();
+  const { loggedInUser, updateLoginContext } = useAuth();
+  const { updateSnackBarMessage } = useSnackBar();
+
+  const handleEditSubmit = (
+    { newEmail, newUsername }: { newEmail: string; newUsername: string },
+    { setSubmitting }: FormikHelpers<{ newEmail: string; newUsername: string }>,
+  ) => {
+    updateProfile(newUsername, newEmail).then((data) => {
+      if (data.error) {
+        console.error({ error: data.error });
+        setSubmitting(false);
+        updateSnackBarMessage(data.error);
+      } else if (data.success) {
+        updateLoginContext(data.success);
+        updateSnackBarMessage('Profile Edited');
+      } else {
+        // should not get here from backend but this catch is for an unknown issue
+        console.error({ data });
+        setSubmitting(false);
+        updateSnackBarMessage('An unexpected error occurred. Please try again');
+      }
+    });
+  };
+
+  const handlePasswordSubmit = (
+    { oldPassword, newPassword }: { oldPassword: string; newPassword: string },
+    { setSubmitting }: FormikHelpers<{ oldPassword: string; newPassword: string }>,
+  ) => {
+    changePassword(oldPassword, newPassword).then((data) => {
+      if (data.error) {
+        console.error({ error: data.error });
+        setSubmitting(false);
+        updateSnackBarMessage(data.error);
+      } else if (data.success) {
+        setSubmitting(false);
+        updateSnackBarMessage('Success');
+      } else {
+        // should not get here from backend but this catch is for an unknown issue
+        console.error({ data });
+        setSubmitting(false);
+        updateSnackBarMessage('An unexpected error occurred. Please try again');
+      }
+    });
+  };
+
+  const history = useHistory();
+
+  if (loggedInUser === undefined) return <CircularProgress />;
+  if (!loggedInUser) {
+    history.push('/login');
+    // loading for a split seconds until history.push works
+    return <CircularProgress />;
+  }
```

Comment:
We won't need to do this when we implement `ProtectedRoute`s - can you either remove it or leave a comment to remind about this?

Reply:
I Will leave a comment, will add it in during merge into main as the protected route is already implemented.